Backporting for LTS 2.528.1 RC

lemeurherve opened the pull request:
```
Postponed
---------

JENKINS-16341           Major                   2.529
        Memory leak in createJavaScriptProxy
        https://issues.jenkins.io/browse/JENKINS-16341

JENKINS-76075           Minor                   2.529
        Optimise LogRotator working in weak BuildReference mode
        https://issues.jenkins.io/browse/JENKINS-76075

Fixed
-----

JENKINS-76135           Minor                   2.529
        Jenkins does not compile with eclipse
        https://issues.jenkins.io/browse/JENKINS-76135

JENKINS-76114           Minor                   2.529
        Setup wizard plugin list does not show All/None/Suggested (regression in 2.527)
        regression
        https://issues.jenkins.io/browse/JENKINS-76114

JENKINS-76002           Minor                   2.529
        String comparison in JS when numbers are expected
        https://issues.jenkins.io/browse/JENKINS-76002
```

Ref:
- https://github.com/jenkins-infra/release/issues/762

### Testing done

Compiled with:
```
mvn -am -pl war,bom -Pquick-build clean install
```

### Proposed changelog entries

- N/A

### Proposed changelog category

/label into-lts

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [-] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [-] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [-] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [-] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@MarkEWaite @daniel-beck @jtnord @jeromepochat 

Before the changes are marked as `ready-for-merge`:

### Maintainer checklist

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

jeromepochat committed:
[JENKINS-76002] Apply number instead of string comparison (#11059)

(cherry picked from commit 88dd0bf930b022781ed73918571bcfa9976d55c1)

jeromepochat committed:
[JENKINS-76114] Bring back the "All/None/Suggested" buttons in the plugin configuration wizard (#11070)

Fix padding

(cherry picked from commit 02b40b1dd150427b49f7513c18947aec274b41e4)

jtnord committed:
[JENKINS-76135] work around eclipse compiler bug (#11087)

https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2860
prevents the code from compiling in eclipse.

Move the class out so that eclipse is happy.  As the class was private
there should be no impact to anything.

(cherry picked from commit c23cf20ea2b923ba7a61e29523df6a9b0913688a)

timja commented:
Is there a reason:

```
JENKINS-16341           Major                   2.529
        Memory leak in createJavaScriptProxy
        https://issues.jenkins.io/browse/JENKINS-16341
```

was postponed?

timja commented:
See https://github.com/apache/maven-javadoc-plugin/issues/1241#issuecomment-3214956373

@daniel-beck hit this in the security release IIRC correctly.

cc @jglick too who is on that issue

looks like setting `detectOfflineLinks=false` is the recommended fix.

lemeurherve committed:
Add `detectOfflineLinks` set to `false` in `maven-javadoc-plugin` configuration

See https://github.com/apache/maven-javadoc-plugin/issues/1241#issuecomment-3214956373

Avoid the following error:
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar (attach-javadocs) on project jenkins-war: Execution attach-javadocs of goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar failed: Null charset name -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar (attach-javadocs) on project jenkins-war: Execution attach-javadocs of goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar failed: Null charset name

lemeurherve committed:
fix: add `-DdetectOfflineLinks=false` tp `mvn` parameter

Avoid the following error:
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar (attach-javadocs) on project jenkins-war: Execution attach-javadocs of goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar failed: Null charset name -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar (attach-javadocs) on project jenkins-war: Execution attach-javadocs of goal org.apache.maven.plugins:maven-javadoc-plugin:3.11.3:jar failed: Null charset name

lemeurherve commented:
Pushed 79743a1c22c215a6538ee2e89c2d3a3796e8e470 in this PR.

lemeurherve committed:
Revert "fix: add `-DdetectOfflineLinks=false` tp `mvn` parameter"

This reverts commit 79743a1c22c215a6538ee2e89c2d3a3796e8e470.

lemeurherve commented on Jenkinsfile:
```diff
@@ -101,6 +101,7 @@ axes.values().combinations {
               'help:evaluate',
               '-Dexpression=changelist',
               "-Doutput=$changelistF",
+              '-DdetectOfflineLinks=false',
```
- Reverted 79743a1c22c215a6538ee2e89c2d3a3796e8e470 and cherry-picked 742357c46c7300ed954a77d9e7f8a54fe8edc1f5 from #11132

lemeurherve committed:
restore `<node.version>24.7.0</node.version>`

lemeurherve commented:
Windows and Linux JDK17 builds passed in https://ci.jenkins.io/job/Core/job/jenkins/job/PR-11130/10/ 🎉 

Unfortunately, the Linux JDK21 agent has been lost on the way: https://ci.jenkins.io/job/Core/job/jenkins/job/PR-11130/10/pipeline-overview/?start-byte=41954&selected-node=332#log-332-1162
> jnlp-maven-21-tjffc seems to be removed or offline (hudson.remoting.ChannelClosedException: Channel "hudson.remoting.Channel@1eec5df4:JNLP4-connect connection from 3.18.21.229/3.18.21.229:9129": Remote call on JNLP4-connect connection from 3.18.21.229/3.18.21.229:9129 failed. The channel is closing down or has closed down); will wait for 5 min 0 sec for it to come back online

MarkEWaite reviewed (approved):
Ready to merge for Wednesday release candidate.

Test failures are related to infrastructure, not the code.

lemeurherve commented:
Got the incrementals from a replay with Linux only: https://github.com/jenkinsci/jenkins/pull/11130/checks?check_run_id=51653704779
> Deployed version 2.528.1-rc37451.ceb_97a_b_cb_f72 to Incrementals

Opened ath & bom draft PRs:
- https://github.com/jenkinsci/acceptance-test-harness/pull/2192
- https://github.com/jenkinsci/bom/pull/5770

MarkEWaite committed:
Merge branch 'stable-2.528' into stable-2.528.1-backport-1

MarkEWaite commented:
I merged the stable-2.528 branch into this branch so that we can see the build results before merging it.  Not strictly required, since I've checked this pull request successfully multiple times on my home Jenkins setup, but it will give us further confidence that ci.jenkins.io and this branch are both healthy.

* [![Build Status](https://ci.jenkins.io/job/Core/job/jenkins/job/stable-2.528/badge/icon?subject=stable-2.528+branch)](https://ci.jenkins.io/job/Core/job/jenkins/job/stable-2.528/)
* [![Build Status](https://ci.jenkins.io/job/Core/job/jenkins/view/change-requests/job/PR-11130/badge/icon?subject=Backport+PR+11130)](https://ci.jenkins.io/job/Core/job/jenkins/view/change-requests/job/PR-11130/)
